src/curlpp/Easy.cpp

```diff
@@ -64,10 +64,10 @@ curlpp::Easy::setOpt(const OptionBase & option)
 
 
 void
-curlpp::Easy::setOpt(std::auto_ptr<OptionBase> option)
+curlpp::Easy::setOpt(std::shared_ptr<OptionBase> option)
 {
 	option->updateHandleToMe(mCurl.get());
-	mOptions.setOpt(option.release());    
+	mOptions.setOpt(option.get());    
```

Comment:
Here, mOptions is a structure containing naked pointers. At its destruction it will delete the pointers it contains. This means that shared_ptr still pointing to that value will have their associated memory deleted. 

So, this change should modify those std::auto_ptr to std::unique_ptr